src/librustdoc/clean/mod.rs

```diff
@@ -1809,59 +1753,60 @@ impl Clean<Visibility> for hir::Visibility<'_> {
         match self.node {
```

Comment:
I think you can actually remove `impl Clean<Visibility> for hir::Visibility<'_>` entirely?
It's now used only for resetting visibilities to inherited, so `visibility: Visibility::Inherited` can be used everywhere instead of `visibility: self.vis.clean(cx)`.

Reply:
This is also used for visibilities on imports, which are not always inherited (e.g. if the import is being inlined): https://github.com/rust-lang/rust/blob/fff92031402e88256dbd828dcb2f00bcd2f03df2/src/librustdoc/clean/mod.rs#L2243

That can't use `from_def_id_and_kind` because it makes up a fake DefId. There are also a few other places that aren't using `Inherited`:

https://github.com/rust-lang/rust/blob/fff92031402e88256dbd828dcb2f00bcd2f03df2/src/librustdoc/clean/mod.rs#L2096
https://github.com/rust-lang/rust/blob/fff92031402e88256dbd828dcb2f00bcd2f03df2/src/librustdoc/clean/mod.rs#L2152
https://github.com/rust-lang/rust/blob/fff92031402e88256dbd828dcb2f00bcd2f03df2/src/librustdoc/clean/mod.rs#L2223